src/main/java/com/aws/greengrass/authorization/WildcardTrie.java

```diff
@@ -141,6 +145,14 @@ private WildcardTrie add(String subject, boolean isTerminal) {
                 }
                 return current.add(subject.substring(i + 1), true);
             }
+            if (currentChar == escapeChar) {
+                char actualChar = getActualChar(subject.substring(i));
+                if (actualChar != nullChar) {
+                    sb.append(actualChar);
+                    i = i + 3;
+                    continue;
+                }
+            }
```

Comment:
you have AuthModule tests that validate resources that end with `$` but you don't have any tests that validate the WildCardTrie if it ends in $/if the `actualChar == null`

Reply:
Are you saying something like 
add("abc$") should match "abc$"

Reply:
`$` is a valid MQTT topic character. For example `$aws`. How does this affect customers currently allowing subscription/publishing on $aws?

Reply:
Only `$` is still valid, its only when the resource has `$` followed by `{` we start checking if its using escape sequence correctly.

Reply:
But it is intended have special meaning, such that if I wanted a literal $, I would put ${$}

Reply:
That makes sense as we have escaping allowed for `$`. But acc to IoT Core policy's behavior. A `$` doesn't always need needs escaping and we are trying to match that.